src/main/java/pl/noip/lolstats/lol/stats/controller/SummonerMatchesController.java

```diff
@@ -0,0 +1,58 @@
+package pl.noip.lolstats.lol.stats.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.noip.lolstats.lol.stats.Exceptions.BearerNotPresentException;
+import pl.noip.lolstats.lol.stats.dto.Match;
+import pl.noip.lolstats.lol.stats.dto.SummonerMatchesResponse;
+import pl.noip.lolstats.lol.stats.jwt.JwtParser;
+import pl.noip.lolstats.lol.stats.jwt.TokenSplit;
+import pl.noip.lolstats.lol.stats.service.RiotRestClient;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/summoner/matches")
```

Comment:
Only summoners play matches, so maybe set path to sth like api/riot/matches. Also make one controller class for every riot api call